src/shared/ui/picker.ts

```diff
@@ -145,3 +149,201 @@ export function verifySinglePickerOutput<T extends vscode.QuickPickItem>(choices
 
     return choices[0]
 }
+
+// TODO: Cache these results? Should we have a separate store? Can we also use a store with values from the explorer tree?
+// If we move this to a cache, we should remove the awsCallLogic and instead listen/send events to/from the cache.
+// (this should also be retooled to not use the request/response objects directly)
+export class IteratingAWSCallPicker<TRequest, TResponse> {
```

Comment:
A few suggestions:

* Separate out the state from this class along with the logic that modifies that state
  * Treat this is as if it were a separate component/service. Events in the UI trigger calls to that service. The UI then converts the response from service into the appropriate format. The UI doesn't know how the service works. The service doesn't know how the UI presents its data. 
* Retain this class to handle UI and delegate to the logic portion
  * Create specific event handlers for each button rather than reusing the same one
* Remove the generalizations of the options and behavior that you don't intend to use (don't add potential future features until necessary).

You should then be able to test the logic / state modifications without the UI. You can also use UI fakes and logic mocks to easily test the UI is performing as expected.